myhomework2.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+
+// не звертайте на цю функцію уваги
+// вона потрібна для того щоб правильно зчитати вхідні дані
+function readHttpLikeInput()
+{
+    $f = fopen('php://stdin', 'r');
+    $store = "";
+    $toread = 0;
+
+    while ($line = fgets($f)) {
+        $store .= preg_replace("/\r/", "", $line);
+
+        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+            $toread = $m[1] * 1;
+
+        if ($line === PHP_EOL)
+
+            break;
+    }
+
+    if ($toread > 0)
+        $store .= fread($f, $toread);
+
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+function parseTcpStringAsHttpRequest($string)
+{
+
+    return [
+        "method" => getMethod($string),
+        "uri" => getUri($string),
+        "headers" => getHeaders($string),
+        "body" => getBody($string),
+    ];
+}
+
+function getHeaders($string)
+{
+    $lines = explode("\n", $string);
+    $headers = [];
+
+    for ($i = 1; $i < sizeof($lines); $i++) {
```

Comment:
`sizeof` is alias of `count`. better to use `count`, because sizeof looks like it will return amount of memory allocated for the variable

Reply:
also, sizeof will be called on each iteration. that's not good